systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java

```diff
@@ -0,0 +1,328 @@
+        String kafkaClientsPodName = kubeClient().listPodsByPrefixInName(kafkaClientsName).get(0).getMetadata().getName();
+
+        KafkaConnectorResource.create(KafkaConnectorResource.kafkaConnector(clusterName)
+            .editOrNewSpec()
+                .withClassName(ECHO_SINK_CLASS_NAME)
+                .withConfig(connectorConfig)
+            .endSpec()
+            .build());
+
+        KafkaConnector kafkaConnector = KafkaConnectorResource.kafkaConnectorClient().inNamespace(NAMESPACE).withName(clusterName).get();
+
+        assertThat(kafkaConnector.getSpec().getClassName(), is(ECHO_SINK_CLASS_NAME));
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+            .withUsingPodName(kafkaClientsPodName)
+            .withTopicName(topicName)
+            .withNamespaceName(NAMESPACE)
+            .withClusterName(clusterName)
+            .withMessageCount(MESSAGE_COUNT)
+            .withListenerName(Constants.PLAIN_LISTENER_DEFAULT_NAME)
+            .build();
+
+        internalKafkaClient.sendMessagesPlain();
+
+        String connectPodName = kubeClient().listPodNames(Labels.STRIMZI_KIND_LABEL, KafkaConnect.RESOURCE_KIND).get(0);
+        PodUtils.waitUntilMessageIsInPodLogs(connectPodName, "Received message with key 'null' and value '99'");
+    }
+
+    @OpenShiftOnly
+    @Test
+    void testPushIntoImageStream() {
+        String imageStreamName = "custom-image-stream";
+        ImageStream imageStream = new ImageStreamBuilder()
+            .editOrNewMetadata()
+                .withName(imageStreamName)
+                .withNamespace(NAMESPACE)
+            .endMetadata()
+            .build();
+
+        kubeClient().getClient().adapt(OpenShiftClient.class).imageStreams().create(imageStream);
+
+        KafkaConnectResource.create(KafkaConnectResource.kafkaConnect(clusterName, 1)
+            .editOrNewSpec()
+                .withNewBuild()
+                    .withPlugins(PLUGIN_WITH_TAR_AND_JAR)
+                    .withNewImageStreamOutput()
+                        .withNewImage(imageStreamName + ":latest")
+                    .endImageStreamOutput()
+                .endBuild()
+            .endSpec()
+            .build(), false);
+
+        KafkaConnect kafkaConnect = KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(clusterName).get();
+
+        LOGGER.info("Checking, if KafkaConnect has all artifacts and if is successfully created");
+        assertThat(kafkaConnect.getSpec().getBuild().getPlugins().get(0).getArtifacts().size(), is(2));
+        assertThat(kafkaConnect.getSpec().getBuild().getOutput().getType(), is("imagestream"));
+        assertThat(kafkaConnect.getSpec().getBuild().getOutput().getImage(), is(imageStreamName + ":latest"));
+        assertThat(kafkaConnect.getStatus().getConditions().get(0).getType(), is(Ready.toString()));
```

Comment:
I think you need to wait for readiness of the connect here and check that the plugin is added to the status. Or does the create wait for readiness?

Reply:
How I mentioned "above" and you wrote here, the readiness wait is inside the create method.

Reply:
Maybe we could still check the list of plugins in status?

Reply:
Yes, good idea, thanks.